crates/consensus/src/transaction/mod.rs

```diff
@@ -84,6 +84,9 @@ pub trait SignableTransaction<Signature>: Transaction {
     /// RLP-encodes the transaction for signing.
     fn encode_for_signing(&self, out: &mut dyn alloy_rlp::BufMut);
 
+    /// Transaction Type
+    fn tx_type(&self) -> TxType;
```

Comment:
Unfortunately using `TxType` in this interface would require network paramterization. Instead I think we could make this function signature `fn use_eip155(&self) -> bool`

However, i'm also concerned about users who intentionally want a non-eip155 on a legacy tx (e.g. people using Nick's method to deploy deterministic contract addresses across networks). If eip-155 is always applied, they would be 

Maybe the long-term solution is to make an `Option<chain_id>` prop on the `LegacyTransaction`, just for signaling eip155 (non-)usage? That prop could be lost when RLPing, but that seems ok I guess? 🤔

Reply:
cc @mattsse

Reply:
> Unfortunately using TxType in this interface would require network paramterization. Instead I think we could make this function signature fn use_eip155(&self) -> bool

That will work too. 
But generally because Alloy is meant to be used by other users (other than foundry) I think maybe adding a tx_type in the Transaction trait might even be the better option. 
Though I did not understand what is exactly meant by "network paramterization". Isn't the type already a part of transaction?

> Maybe the long-term solution is to make an Option<chain_id> prop on the LegacyTransaction, just for signaling eip155 (non-)usage? That prop could be lost when RLPing, but that seems ok I guess?

I agree. I think ganache also behaved like this. When there was no chain id value in a legacy transaction it would not do a EIP155 encoding.

Reply:
Sorry to ping you guys. Any updates?

cc @mattsse